core/score_sync.go

```diff
@@ -0,0 +1,101 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"application/core/model"
+	"log"
+	"os"
+	"strconv"
+)
+
+// SyncAmgUUIDForScore fetches the AmgUUID from Core BB and populates the external_user_id field
+func (app *Application) SyncAmgUUIDForScore(score *model.Score, externalIDs map[string]string) error {
+	// Get Mastodon ID from externalIDs if provided, otherwise use the one already in score
+	mastodonID := score.ExternalProfileID
+	if externalIDs != nil {
+		if mastodonIDFromMap, ok := externalIDs["mastodon_id"]; ok && mastodonIDFromMap != "" {
+			mastodonID = mastodonIDFromMap
+		}
+	}
+
+	if mastodonID == "" {
+		log.Printf("SyncAmgUUIDForScore: no mastodon_id provided for score, skipping AmgUUID sync")
+		return nil
+	}
+
+	// Look up the AmgUUID from Core BB
+	accountCriteria := map[string]interface{}{
+		"identifiers": map[string]interface{}{
+			"$elemMatch": map[string]interface{}{
+				"code":       "mastodon_id",
+				"identifier": mastodonID,
+			},
+		},
+	}
+
+	coreAccounts, err := app.corebb.RetrieveCoreUserAccountByCriteria(accountCriteria, &score.AppID, &score.OrgID)
+	if err != nil {
+		log.Printf("SyncAmgUUIDForScore: error retrieving Core account: %v", err)
+		// Don't fail the score creation if Core BB lookup fails, just log it
+		return nil
+	}
+
+	if len(coreAccounts) == 0 {
+		log.Printf("SyncAmgUUIDForScore: no Core account found for mastodon_id %s", mastodonID)
+		return nil
+	}
+
+	if len(coreAccounts) > 1 {
+		log.Printf("SyncAmgUUIDForScore: WARNING - multiple Core accounts found for mastodon_id %s, using first one", mastodonID)
+	}
+
+	// Set the AmgUUID
+	score.ExternalUserID = coreAccounts[0].ID
+	log.Printf("SyncAmgUUIDForScore: synced AmgUUID %s for mastodon_id %s", score.ExternalUserID, mastodonID)
+
+	return nil
+}
+
+// PrepareScoreForResponse modifies a score for API response based on feature flag
+func (app *Application) PrepareScoreForResponse(score *model.Score) {
+	if app.shouldUseExternalUserID() && score.ExternalUserID != "" {
+		// When flag is enabled and we have an AmgUUID, use it in the external_profile_id field
+		score.ExternalProfileID = score.ExternalUserID
+	}
+}
+
+// PrepareScoresForResponse prepares multiple scores for API response
+func (app *Application) PrepareScoresForResponse(scores []model.Score) {
+	for i := range scores {
+		app.PrepareScoreForResponse(&scores[i])
+	}
+}
+
+// shouldUseExternalUserID reads the feature flag from environment variables
+func (app *Application) shouldUseExternalUserID() bool {
+	useExternalUserIDStr := os.Getenv("USE_EXTERNAL_USER_ID")
+	if useExternalUserIDStr == "" {
+		return false // Default to false
+	}
+
+	useExternalUserID, err := strconv.ParseBool(useExternalUserIDStr)
+	if err != nil {
+		log.Printf("Warning: invalid USE_EXTERNAL_USER_ID value '%s', defaulting to false", useExternalUserIDStr)
+		return false
+	}
+
+	return useExternalUserID
+}
```

Comment:
Environment variable reading and parsing should be done in `main.go` so that it is all in one place.